python/pyspark/sql/functions.py

```diff
@@ -1437,10 +1453,6 @@ def hash(*cols):
     'ascii': 'Computes the numeric value of the first character of the string column.',
     'base64': 'Computes the BASE64 encoding of a binary column and returns it as a string column.',
     'unbase64': 'Decodes a BASE64 encoded string column and returns it as a binary column.',
-    'initcap': 'Returns a new string column by converting the first letter of each word to ' +
-               'uppercase. Words are delimited by whitespace.',
-    'lower': 'Converts a string column to lower case.',
-    'upper': 'Converts a string column to upper case.',
     'ltrim': 'Trim the spaces from left end for the specified string value.',
     'rtrim': 'Trim the spaces from right end for the specified string value.',
     'trim': 'Trim the spaces from both ends for the specified string column.',
```

Comment:
Did you not add the column name support because Scala side has this signautre below?:

```scala
  def trim(e: Column): Column
  def trim(e: Column, trimString: String): Column
```

That's not allowed in Python

```python
>>> from pyspark.sql.functions import trim, lit
>>> spark.range(1).select(lit('a').alias("value")).select(trim("value", "a"))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: _() takes exactly 1 argument (2 given)
```

Reply:
This is what I initially expected when I asked to whitelist them.

Reply:
I don't understand what you're saying here. `trim` with 2 arguments was never supported in PySpark, and that's a separate issue. What this patch changes is that `trim("value")` is now supported, when it wasn't previously.

Reply:
I was saying this because I didn't get why you excluded string functions.